rs/execution_environment/src/canister_manager.rs

```diff
@@ -62,6 +64,18 @@ use std::{convert::TryFrom, str::FromStr, sync::Arc};
 use types::*;
 pub(crate) mod types;
 
+/// The maximum size of a slice of binary data that a user
+/// can download in one mwessage.
+const MAX_SLICE_SIZE_BYTES: u64 = 2_000_000;
+
+// A slice must fit into the maximum 2MiB message size, with
```

Comment:
same here: `// Ensure the slice, with extra room for Candid encoding, fits within 2 MiB.`